docs/DeveloperGuide.md

```diff
@@ -1,38 +1,442 @@
 
-## Glossary
 
-* *glossary item* - Definition
+---
 
-## Instructions for manual testing
 
-{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+# Setting up, getting started
+
+
+## Setting up the project in your computer
+
+First, fork this repo, and clone the fork into your computer.
+
+If you plan to use Intellij IDEA (highly recommended):
+
+
+
+1. **Configure the JDK:** Follow the guide [[se-edu/guides] IDEA: Configuring the JDK](https://se-education.org/guides/tutorials/intellijJdk.html) to to ensure Intellij is configured to use JDK 11.
+2. **Import the project as a Gradle project:** Follow the guide [[se-edu/guides] IDEA: Importing a Gradle project](https://se-education.org/guides/tutorials/intellijImportGradleProject.html) to import the project into IDEA. \
+
+<img src="./DGDiagramsCreator/DGUsedDiagrams/ExclamationMark.png" width="4%" />
+Note: Importing a Gradle project is slightly different from importing a normal Java project.
+3. **Verify the setup:**
+    1. Run the _seedu.address.Main_ and try a few commands.
+    2. [Run the tests](https://se-education.org/addressbook-level3/Testing.html) to ensure they all pass.
+
+
+## Before writing code
+
+
+
+1. **Configure the coding style** \
+   If using IDEA, follow the guide [[se-edu/guides] IDEA: Configuring the code style](https://se-education.org/guides/tutorials/intellijCodeStyle.html) to set up IDEA’s coding style to match ours. 
+
+<img src="./DGDiagramsCreator/DGUsedDiagrams/ExclamationMark.png" width="4%" />
+Tip: Optionally, you can follow the guide [[se-edu/guides] Using Checkstyle](https://se-education.org/guides/tutorials/checkstyle.html) to find how to use the CheckStyle within IDEA e.g., to report problems as you write code.  
+
+2. **Set up CI** This project comes with a GitHub Actions config files (in .github/workflows folder). When GitHub detects those files, it will run the CI for your project automatically at each push to the master branch or to any PR. No set up required.  
+3. **Learn the design** When you are ready to start coding, we recommend that you get some sense of the overall design by reading about [AddressBook’s architecture](https://se-education.org/addressbook-level3/DeveloperGuide.html#architecture).  
+4. **Do the tutorials These tutorials will help you get acquainted with the codebase.**  
+    * [Tracing code](https://se-education.org/addressbook-level3/tutorials/TracingCode.html)
+    * [Adding a new command](https://se-education.org/addressbook-level3/tutorials/AddRemark.html)
+    * [Removing fields](https://se-education.org/addressbook-level3/tutorials/RemovingFields.html)
+
+
+---
+
+[//]: # (@@author ollayf)
+# Design & Implementation
+
+
+## Architecture
+
+<img src="./DGDiagramsCreator/DGUsedDiagrams/ArchitectureDiagram.png" width="100%" />
+
+
+This **Architecture Diagram** explains the high-level design of the App.
```

Comment:
Perhaps use the standard CS2113 representation for diagrams? (instead of using the symbols)